src/cursor/cur_file.c

```diff
@@ -693,6 +693,10 @@ __curfile_create(WT_SESSION_IMPL *session,
 		cacheable = false;
 	}
 
+	WT_ERR(__wt_config_gets_def(session, cfg, "read_once", 0, &cval));
+	if (cval.val != 0)
+		F_SET(cbt, WT_CBT_READ_WONT_NEED);
```

Comment:
I'm still investigating exactly what it takes to address this, but the patch as shown, to my knowledge only "works" if `read_once` is passed into `open_cursor` specifically for a `file:` URI or a simple `table:` URI. I suspect when higher level cursors create a lower level btree cursor, the configuration is not forwarded down? Still investigating.

Reply:
Concern withdrawn. I can see the configuration being passed down.